awx_collection/plugins/modules/tower_project.py

```diff
@@ -212,6 +219,7 @@ def main():
                                         custom_virtualenv=custom_virtualenv,
                                         create_on_missing=True)
                 json_output['id'] = result['id']
+                project.wait(pk=None, parent_pk=result['id'])
```

Comment:
Oh, I think you need to make use of the `wait` variable you previously saved, and not call this if it's not true